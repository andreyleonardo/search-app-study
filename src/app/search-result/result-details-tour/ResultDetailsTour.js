import React from 'react';
import PropTypes from 'prop-types';
import he from 'he';
import classnames from 'classnames';

import './ResultDetailsTour.scss';

const ResultDetailsTour = ({
  index, description, expand, imgAlt, imgSrc, title
}) => (
  <div
    id={`result-details-tour-id-${index}`}
    aria-expanded={expand}
    className={classnames('result_details_tour row', { 'result_details_tour--expanded': expand })}
  >
    <picture className="result_details_tour__picture hidden-xs col-sm-12">
      <img className="result_details_tour__img" src={imgSrc} alt={imgAlt} />
    </picture>
    <article className="result_details_tour__description col-xs-12">
      <h1>
        {he.decode(title)}
      </h1>
      <p>{he.decode(description)}</p>
    </article>
  </div>
);

ResultDetailsTour.propTypes = {
  title: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  expand: PropTypes.bool,
  imgAlt: PropTypes.string,
  imgSrc: PropTypes.string.isRequired,
  index: PropTypes.number
};

ResultDetailsTour.defaultProps = {
  expand: false,
  imgAlt: '',
  index: 0
};

export default ResultDetailsTour;
