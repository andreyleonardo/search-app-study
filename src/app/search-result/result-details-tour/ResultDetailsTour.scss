@import "src/stylesheets/css/_colors";

.result_details_tour {
  background-color: $primary;
  color: white;
  flex-basis: 100%;
  max-height: 0px;
  overflow: hidden;
  margin: 3px;
  margin-top: -3px;
  transition: all .3s ease-in-out;

  &--expanded {
    padding: 0.2em;
    max-height: 1000px;
  }

  &__img {
    width: 100%;
    height: 100%;
    object-fit: cover;
  }

  &__picture {
    padding: 10px;
  }

  &__description {
    height: 100%;
  }
}
