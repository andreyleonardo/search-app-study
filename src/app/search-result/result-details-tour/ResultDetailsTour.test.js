import React from 'react';
import ResultDetailsTour from './ResultDetailsTour';

describe('Result Details Tour', () => {
  let resultCardTour;
  let props;

  beforeEach(() => {
    props = {
      description: 'Tour Title',
      expand: false
    };

    resultCardTour = shallow(<ResultDetailsTour {...props} />);
  });

  it('renders without crashing', () => {
    expect(resultCardTour).toMatchSnapshot();
  });

  it('sets expanded class when expand is true', () => {
    resultCardTour.setProps({ expand: true });

    expect(resultCardTour.exists('.result_details_tour--expanded')).toBeTruthy();
  });
});
