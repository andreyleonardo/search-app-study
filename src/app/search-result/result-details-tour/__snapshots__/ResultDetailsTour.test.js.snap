// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Result Details Tour renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ResultDetailsTour
    description="Tour Description"
    expand={false}
    imgAlt=""
    imgSrc="image source"
    index={0}
    title="Tour title"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "aria-expanded": false,
      "children": Array [
        <picture
          className="result_details_tour__picture hidden-xs col-sm-12"
        >
          <img
            alt=""
            className="result_details_tour__img"
            src="image source"
          />
        </picture>,
        <article
          className="result_details_tour__description col-xs-12"
        >
          <h1>
            Tour title
          </h1>
          <p>
            Tour Description
          </p>
        </article>,
      ],
      "className": "result_details_tour row",
      "id": "result-details-tour-id-0",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <img
            alt=""
            className="result_details_tour__img"
            src="image source"
          />,
          "className": "result_details_tour__picture hidden-xs col-sm-12",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "",
            "className": "result_details_tour__img",
            "src": "image source",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        "type": "picture",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1>
              Tour title
            </h1>,
            <p>
              Tour Description
            </p>,
          ],
          "className": "result_details_tour__description col-xs-12",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Tour title",
            },
            "ref": null,
            "rendered": "Tour title",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Tour Description",
            },
            "ref": null,
            "rendered": "Tour Description",
            "type": "p",
          },
        ],
        "type": "article",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "aria-expanded": false,
        "children": Array [
          <picture
            className="result_details_tour__picture hidden-xs col-sm-12"
          >
            <img
              alt=""
              className="result_details_tour__img"
              src="image source"
            />
          </picture>,
          <article
            className="result_details_tour__description col-xs-12"
          >
            <h1>
              Tour title
            </h1>
            <p>
              Tour Description
            </p>
          </article>,
        ],
        "className": "result_details_tour row",
        "id": "result-details-tour-id-0",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <img
              alt=""
              className="result_details_tour__img"
              src="image source"
            />,
            "className": "result_details_tour__picture hidden-xs col-sm-12",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "",
              "className": "result_details_tour__img",
              "src": "image source",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": "picture",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1>
                Tour title
              </h1>,
              <p>
                Tour Description
              </p>,
            ],
            "className": "result_details_tour__description col-xs-12",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Tour title",
              },
              "ref": null,
              "rendered": "Tour title",
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Tour Description",
              },
              "ref": null,
              "rendered": "Tour Description",
              "type": "p",
            },
          ],
          "type": "article",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
